nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -0,0 +1,352 @@
+
+        try {
+            AutomationServer srv = Framework.getService(AutomationServer.class);
+            if (!srv.accept(op.getId(), op.isChain(), request)) {
+                return ResponseHelper.notFound();
+            }
+            final String executionId = UUID.randomUUID().toString();
+
+            // XXX: AsyncContext asyncContext = ctx.getRequest().startAsync();
+            new Thread(() -> {
+                try {
+                    op.execute(xreq, new OperationCallback() {
+
+                        @Override
+                        public void onChainEnter(OperationType chain) {
+                            //
+                        }
+
+                        @Override
+                        public void onChainExit() {
+                            setCompleted(executionId);
+                        }
+
+                        @Override
+                        public void onOperationEnter(OperationContext context, OperationType type, InvokableMethod method,
+                                Map<String, Object> params) {
+                            enterMethod(executionId, method);
+                        }
+
+                        @Override
+                        public void onOperationExit(Object output) {
+                            setOutput(executionId, (Serializable) output);
+                        }
+
+                        @Override
+                        public OperationException onError(OperationException error) {
+                            setError(executionId, error.getMessage());
+                            return error;
+                        }
+
+                    });
+                } catch (OperationException e) {
+                    setError(executionId, e.getMessage());
+                }
+            }).run();
+
+            String statusURL = String.format("%s/%s/status", ctx.getURL(), executionId);
+            return Response.status(Response.Status.ACCEPTED).location(new URI(statusURL)).build();
+
+        } catch (URISyntaxException cause) {
+            String exceptionMessage = "Failed to invoke operation: " + op.getId();
+            Throwable unWrapException = ExceptionHelper.unwrapException(cause);
+            if (unWrapException instanceof RestOperationException) {
+                int customHttpStatus = ((RestOperationException) unWrapException).getStatus();
+                throw new NuxeoException(exceptionMessage, cause, customHttpStatus);
+            }
+            throw new NuxeoException(exceptionMessage, cause);
+        }
+    }
+
+    @GET
+    @Path("{executionId}/status")
+    public Object status(@PathParam("executionId") String executionId)
+            throws IOException, URISyntaxException, MessagingException {
+        if (isCompleted(executionId)) {
+            String error = getError(executionId);
+            if (error != null) {
+                throw new NuxeoException(error, Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
```

Comment:
Could be replaced by:
```suggestion
                throw new NuxeoException(error, HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
```